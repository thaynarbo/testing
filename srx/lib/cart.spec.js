import Cart from './cart';

describe('Cart', () => {
    let cart;
    let product = {
        name: 'Adidas Sneaker',
        price: 30075,
    };
    let product2 = {
        name: 'Nike Sneaker',
        price: 45055,
    };

    beforeEach(() => {
        cart = new Cart();
    });

    describe('getTotal()', () => {
        it('should return 0 if there are no items', () => {
            expect(cart.getTotal().getAmount()).toBe(0);
        });

        it('should return sum of all products in the list ', () => {
            const item = {
                product,
                quantity: 2,
            };

            cart.add(item);

            expect(cart.getTotal().getAmount()).toEqual(60150);
        });

        it('should not allow the same product to be add more than once', () => {
            cart.add({
                product,
                quantity: 2,
            });
            cart.add({
                product,
                quantity: 1,
            });

            expect(cart.getTotal().getAmount()).toEqual(30075);
        });

        it('should remove item and receive total taking the new item into consideration', () => {
            cart.add({
                product,
                quantity: 2,
            });

            cart.add({
                product: product2,
                quantity: 1,
            });

            cart.remove(product);

            expect(cart.getTotal().getAmount()).toEqual(45055);
        });

        it('should return total with the discount applied', () => {
            cart.add({
                product,
                quantity: 2,
                condition: {
                    percentage: 30,
                    minimum: 2,
                },
            });

            expect(cart.getTotal().getAmount()).toEqual(42105);
        });

        it('should apply discount base on quantity', () => {
            const condition = {
                quantity: 2,
            };

            cart.add({
                product,
                condition,
                quantity: 4,
            });

            expect(cart.getTotal().getAmount()).toEqual(60150);
        });
    });

    describe('checkout()', () => {
        it('should return an object with total and list of items', () => {
            cart.add({
                product,
                quantity: 2,
            });

            cart.add({
                product: product2,
                quantity: 3,
            });

            expect(cart.checkout()).toMatchSnapshot();
        });
        it('should return an object with total and list of items when summary is called', () => {
            cart.add({
                product,
                quantity: 2,
            });

            cart.add({
                product: product2,
                quantity: 3,
            });

            expect(cart.summary()).toMatchSnapshot();
            expect(cart.getTotal().getAmount()).toBeGreaterThan(0);
        });
    });
});
