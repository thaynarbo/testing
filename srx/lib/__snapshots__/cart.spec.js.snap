// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart checkout() should return an object with total and list of items 1`] = `
Object {
  "items": Array [],
  "total": 195315,
}
`;

exports[`Cart checkout() should return an object with total and list of items when summary is called 1`] = `
Object {
  "items": Array [
    Object {
      "product": Object {
        "name": "Adidas Sneaker",
        "price": 30075,
      },
      "quantity": 2,
    },
    Object {
      "product": Object {
        "name": "Nike Sneaker",
        "price": 45055,
      },
      "quantity": 3,
    },
  ],
  "total": 195315,
}
`;
